shell/browser/electron_navigation_throttle.cc

```diff
@@ -37,10 +59,35 @@ ElectronNavigationThrottle::WillStartRequest() {
     return PROCEED;
   }
 
-  if (handle->IsRendererInitiated() && handle->IsInMainFrame() &&
-      api_contents->EmitNavigationEvent("will-navigate", handle)) {
+  std::shared_ptr<bool> is_delay_called(new bool(false));
+  std::shared_ptr<std::string> error_html(new std::string);
```

Comment:
It would be safe to pass a raw pointer since there is already code checking delayed call.

```c++
std::string error_html;
base::BindOnce(&HandleEventReturnValue, &error_html);
```